cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/MonitorUtils.java

```diff
@@ -436,24 +437,33 @@ static void setBadBrokerState(ClusterModel clusterModel, Cluster cluster) {
    * @param valuesAndExtrapolations The values and extrapolations of the leader replica.
    * @param replicaPlacementInfo The distribution of replicas over broker logdirs if available, {@code null} otherwise.
    * @param brokerCapacityConfigResolver The resolver for retrieving broker capacities.
+   * @param deadBrokers Set of dead brokers in the cluster.
    */
   static void populatePartitionLoad(Cluster cluster,
                                     ClusterModel clusterModel,
                                     TopicPartition tp,
                                     ValuesAndExtrapolations valuesAndExtrapolations,
                                     Map<TopicPartition, Map<Integer, String>> replicaPlacementInfo,
-                                    BrokerCapacityConfigResolver brokerCapacityConfigResolver) {
+                                    BrokerCapacityConfigResolver brokerCapacityConfigResolver,
+                                    Set<Integer> deadBrokers) {
     PartitionInfo partitionInfo = cluster.partition(tp);
     // If partition info does not exist, the topic may have been deleted.
     if (partitionInfo != null) {
       boolean needToAdjustCpuUsage = true;
       for (int index = 0; index < partitionInfo.replicas().length; index++) {
         Node replica = partitionInfo.replicas()[index];
         String rack = getRackHandleNull(replica);
-        // Note that we assume the capacity resolver can still return the broker capacity even if the broker
-        // is dead. We need this to get the host resource capacity.
-        BrokerCapacityInfo brokerCapacity =
-            brokerCapacityConfigResolver.capacityForBroker(rack, replica.host(), replica.id());
+        BrokerCapacityInfo brokerCapacity;
+        try {
+          brokerCapacity = brokerCapacityConfigResolver.capacityForBroker(rack, replica.host(), replica.id());
+        } catch (Exception e) {
+          //Capacity resolver may not be able to return the capacity information of dead brokers.
+          if (deadBrokers.contains(replica.id())) {
+            brokerCapacity = new BrokerCapacityInfo(Collections.emptyMap());
```

Comment:
Passing an empty map to `BrokerCapacityInfo` means that the broker has no resources -- I suspect that this will cause issues in the interpreting logic, which expects a value for each resource type (e.g. populating partition load on such brokers -- also see `Broker#_brokerCapacity`).

1. Can we add a sanity check in `BrokerCapacityInfo` constructors to ensure that any capacity sent to this broker has exactly all resource types in the keyset of map?
2. Can we add a unit test to verify our expected behavior and a cluster model can be generated properly under the tested scenario?